src/libraries/Microsoft.PowerFx.Core/Functions/Delegation/DelegationStrategies/OpDelegationStrategy.cs

```diff
@@ -111,6 +111,12 @@ private bool IsSupportedNode(TexlNode node, OperationCapabilityMetadata metadata
                 return true;
             }
 
+            // Non row-scope, non async, pure nodes should always be valid because we can calculate value in runtime before delegation.
+            if (!binding.IsRowScope(node) && !binding.IsAsync(node) && binding.IsPure(node))
+            {
+                return true;
+            }
```

Comment:
Please add test coverage for this when you bring this change to Power Apps.

Reply:
will do